src/js/tabs/osd.js

```diff
@@ -296,6 +296,73 @@ OSD.initData = function() {
 };
 OSD.initData();
 
+OSD.getVariantForPreview = function(osdData, elementName) {
+    return osdData.displayItems.find(element => element.name === elementName).variant;
+};
+
+OSD.generateAltitudePreview = function(osdData) {
+    const unit = FONT.symbol(osdData.unit_mode === 0 ? SYM.FEET : SYM.METRE);
+    const variantSelected = OSD.getVariantForPreview(osdData, 'ALTITUDE');
+    return `${FONT.symbol(SYM.ALTITUDE)}399${variantSelected === 0? '.7' : ''}${unit}`;
+};
+
+OSD.generateBatteryUsagePreview = function(osdData) {
+    const variantSelected = OSD.getVariantForPreview(osdData, 'MAIN_BATT_USAGE');
+    let value;
+    switch (variantSelected) {
+        case 0:
+            value = FONT.symbol(SYM.PB_START) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL)
+                + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL)
+                + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_END) + FONT.symbol(SYM.PB_EMPTY)
+                + FONT.symbol(SYM.PB_CLOSE);
+            break;
+
+        case 1:
+            value = FONT.symbol(SYM.PB_START) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL)
+                + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_FULL) + FONT.symbol(SYM.PB_END) + FONT.symbol(SYM.PB_EMPTY)
+                + FONT.symbol(SYM.PB_EMPTY) + FONT.symbol(SYM.PB_EMPTY) + FONT.symbol(SYM.PB_EMPTY) + FONT.symbol(SYM.PB_EMPTY)
+                + FONT.symbol(SYM.PB_CLOSE);
+            break;
```

Comment:
I think pulling out those symbols to vars even in the same scope would make it a bit more legible.
```suggestion
    let value;
    const start = FONT.symbol(SYM.PB_START);
    const full = FONT.symbol(SYM.PB_FULL);
    const end = FONT.symbol(SYM.PB_END);
    const empty =  FONT.symbol(SYM.PB_EMPTY);
    const close = FONT.symbol(SYM.PB_CLOSE);
    switch (variantSelected) {
        case 0:
            value = start + full + full + full + full + full + full + full + full + full + end + empty + close;
            break;

        case 1:
            value = start + full + full + full + full + full + end + empty + empty + empty + empty + empty + close;
            break;
```

Reply:
Can beautify further with .repeat(count)

Reply:
I like the idea... But the weekend I'm out... I will do a new PR later